docs: Added Cache Control headers documentation (#918)
front-commerce/developers.front-commerce.com: docs/02-guides/cache-control.mdx

---
title: Cache-Control headers
description:
  This guide explains how you can add Cache-Control headers in your routes in
  Front-Commerce.
tags:
  - routes
  - cache
---

<SinceVersion tag="3.6" />

> <p>{frontMatter.description}</p>

Author: I think the description bring a nice "placeholder sentence" to introduce the guide, although I understand how it might be considered a bit useless. I'll remove it.

> Cache-Control headers are used to enhance the experience of the user by telling
> the browsers to put some of the page in their internal cache. By default,
> Front-Commerce already provides Cache-Control headers on several meaningful
> routes:

Reviewer: ```suggestion
`Cache-Control` headers are implemented to improve user experience by instructing browsers to store parts of the webpage in their internal cache. By default, Front-Commerce already applies `Cache-Control` headers to several important routes:
routes:
```

- Products pages
- Categories pages
- Home page
- `/robots.txt`
- `/sitemaps.xml`

## How to use Cache-Control headers in Front-Commerce

In Front-Commerce, adding Cache-Control headers to route is done by leveraging
on the
[`CacheControl`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/core/services/implementations/CacheControl.ts?ref_type=heads)
service, available from `FrontCommerceApp`.

:::note

To learn more about available services in Front-Commerce, see our

> [documentation](docs/3.x/api-reference/front-commerce-core/services)

Reviewer: ```suggestion
[documentation](/docs/3.x/api-reference/front-commerce-core/services)
```

:::

> In this example, we'll add Cache-Control headers on a "Acme" route.

Reviewer: ```suggestion
In this example, we'll add `Cache-Control` headers on a "Acme" route.
```
